packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts

```diff
@@ -150,6 +150,19 @@ export const isSqlStrategy = (strategy: ModelDataSourceStrategy): strategy is SQ
   );
 };
 
+/**
+ * Provides the data source strategy for a given model
+ * @param ctx Transformer Context
+ * @param typename Model name
+ * @returns ModelDataSourceStrategyDbType
+ */
+export const getStrategyDbTypeFromModel = (ctx: DataSourceStrategiesProvider, typename: string): ModelDataSourceStrategyDbType => {
+  if (isBuiltInGraphqlType(typename)) {
+    return 'DYNAMODB';
+  }
+  return getModelDataSourceStrategy(ctx, typename).dbType;
```

Comment:
This logic is based on the existing [`isSqlModel` function](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts#L136-L142).